pkg/storegateway/batch_series_test.go

```diff
@@ -808,6 +811,233 @@ func TestBlockSeriesChunkRefsSetsIterator_ErrorPropagation(t *testing.T) {
 	assert.ErrorContains(t, iterator.Err(), "test limit exceeded")
 }
 
+func TestLoadingBatchSet(t *testing.T) {
+	type testBlock struct {
+		ulid   ulid.ULID
+		series []seriesEntry
+	}
+
+	block1 := testBlock{
+		ulid:   ulid.MustNew(1, nil),
+		series: generateSeriesChunks(t, 10),
+	}
+
+	block2 := testBlock{
+		ulid:   ulid.MustNew(2, nil),
+		series: generateSeriesChunks(t, 10),
+	}
+
+	toSeriesChunkRefs := func(block testBlock, seriesIndex int) seriesChunkRefs {
+		series := block.series[seriesIndex]
+
+		chunkRefs := make([]seriesChunkRef, len(series.chks))
+		for i, c := range series.chks {
+			chunkRefs[i] = seriesChunkRef{
+				blockID: block.ulid,
+				ref:     series.refs[i],
+				minTime: c.MinTime,
+				maxTime: c.MaxTime,
+			}
+		}
+
+		return seriesChunkRefs{
+			lset:   series.lset,
+			chunks: chunkRefs,
+		}
+	}
+
+	testCases := map[string]struct {
+		existingBlocks      []testBlock
+		setsToLoad          []seriesChunkRefsSet
+		expectedSets        []seriesChunksSet
+		addLoadErr, loadErr error
+		expectedErr         string
+	}{
+		"loads single set": {
+			existingBlocks: []testBlock{block1},
+			setsToLoad: []seriesChunkRefsSet{
+				{series: []seriesChunkRefs{toSeriesChunkRefs(block1, 0), toSeriesChunkRefs(block1, 1)}},
+			},
+			expectedSets: []seriesChunksSet{
+				{series: []seriesEntry{block1.series[0], block1.series[1]}},
+			},
+		},
+		"loads multiple sets": {
+			existingBlocks: []testBlock{block1},
+			setsToLoad: []seriesChunkRefsSet{
+				{series: []seriesChunkRefs{toSeriesChunkRefs(block1, 0), toSeriesChunkRefs(block1, 1)}},
+				{series: []seriesChunkRefs{toSeriesChunkRefs(block1, 2), toSeriesChunkRefs(block1, 3)}},
+			},
+			expectedSets: []seriesChunksSet{
+				{series: []seriesEntry{block1.series[0], block1.series[1]}},
+				{series: []seriesEntry{block1.series[2], block1.series[3]}},
+			},
+		},
+		"loads single set from multiple blocks": {
+			existingBlocks: []testBlock{block1, block2},
+			setsToLoad: []seriesChunkRefsSet{
+				{series: []seriesChunkRefs{toSeriesChunkRefs(block1, 0), toSeriesChunkRefs(block1, 1)}},
```

Comment:
This is not loading from multiple blocks as stated in the test case name. I think we can just change the 2nd entry from block1 to block2.